runtime/queries/nix/highlights.scm

```diff
@@ -1,81 +1,84 @@
  (#is-not? local))
 
 ((identifier) @function.builtin
  (#match? @function.builtin "^(__add|__addErrorContext|__all|__any|__appendContext|__attrNames|__attrValues|__bitAnd|__bitOr|__bitXor|__catAttrs|__compareVersions|__concatLists|__concatMap|__concatStringsSep|__deepSeq|__div|__elem|__elemAt|__fetchurl|__filter|__filterSource|__findFile|__foldl'|__fromJSON|__functionArgs|__genList|__genericClosure|__getAttr|__getContext|__getEnv|__hasAttr|__hasContext|__hashFile|__hashString|__head|__intersectAttrs|__isAttrs|__isBool|__isFloat|__isFunction|__isInt|__isList|__isPath|__isString|__langVersion|__length|__lessThan|__listToAttrs|__mapAttrs|__match|__mul|__parseDrvName|__partition|__path|__pathExists|__readDir|__readFile|__replaceStrings|__seq|__sort|__split|__splitVersion|__storePath|__stringLength|__sub|__substring|__tail|__toFile|__toJSON|__toPath|__toXML|__trace|__tryEval|__typeOf|__unsafeDiscardOutputDependency|__unsafeDiscardStringContext|__unsafeGetAttrPos|__valueSize|abort|baseNameOf|derivation|derivationStrict|dirOf|fetchGit|fetchMercurial|fetchTarball|fromTOML|import|isNull|map|placeholder|removeAttrs|scopedImport|throw|toString)$")
  (#is-not? local))
 
 [
-  (string)
-  (indented_string)
+  (string_expression)
+  (indented_string_expression)
 ] @string
 
 [
-  (path)
-  (hpath)
-  (spath)
+  (path_expression)
+  (hpath_expression)
+  (spath_expression)
 ] @string.special.path
 
-(uri) @string.special.uri
+(uri_expression) @string.special.uri
 
-; boolean
-((identifier) @constant.builtin.boolean (#match? @constant.builtin.boolean "^(true|false)$")) @constant.builtin.boolean
-; null
-((identifier) @constant.builtin (#eq? @constant.builtin "null")) @constant.builtin
-
-(integer) @constant.numeric.integer
-(float) @constant.numeric.float
+(integer_expression) @constant.numeric.integer
+(float_expression) @constant.numeric.float
 
 (interpolation
   "${" @punctuation.special
   "}" @punctuation.special) @embedded
 
 (escape_sequence) @constant.character.escape
 
-(function
+(function_expression
   universal: (identifier) @variable.parameter
 )
 
 (formal
   name: (identifier) @variable.parameter
   "?"? @punctuation.delimiter)
 
-(app
+(select_expression
+  attrpath: (attrpath (identifier)) @variable.other.member)
+
+(apply_expression
   function: [
-    (identifier) @function
-    (select
+    (variable_expression (identifier)) @function
+    (select_expression
       attrpath: (attrpath
-        attr: (attr_identifier) @function .))])
-
+        attr: (identifier) @function .))])
 
-(unary
+(unary_expression
   operator: _ @operator)
 
-(binary
+(binary_expression
   operator: _ @operator)
 
-(attr_identifier) @variable.other.member
-(inherit attrs: (attrs_inherited (identifier) @variable.other.member) )
+(variable_expression (identifier) @variable)
+
+(binding
+  attrpath: (attrpath (identifier)) @variable.other.member)
+
+(identifier) @variable.other.member
```

Comment:
Oh actually I didn't see the `(identifier) @variable` at the bottom of the file. I think we can just remove this one even though it would be a departure from the upstream highlights.

Reply:
are you saying to remove all 4 lines here, or just line 75?

Reply:
Just L75. That `binding` capture above looks good